torch_patches/X10-device_test.diff

```diff
@@ -1,117 +1,35 @@
 diff --git a/test/common_device_type.py b/test/common_device_type.py
-index 2b47282a5f..a00f80797c 100644
+index dda0b8d791..cee2921bd2 100644
 --- a/test/common_device_type.py
 +++ b/test/common_device_type.py
-@@ -4,6 +4,8 @@ import unittest
- import torch
- from common_utils import TestCase, TEST_WITH_ROCM, TEST_MKL, \
-     skipCUDANonDefaultStreamIf
-+import torch_xla
-+assert torch_xla  # silences flake8
- 
- # Note: Generic Device-Type Testing
- #
-@@ -113,10 +115,118 @@ class CUDATestBase(DeviceTypeTestBase):
+@@ -156,10 +156,71 @@ class CUDATestBase(DeviceTypeTestBase):
          cls.no_magma = not torch.cuda.has_magma
- 
- 
+
+
++import torch_xla
++assert torch_xla  # Silences Flake (unused import)
++
++# Acquires XLA metadata
++import os
++from runpy import run_path
++file_dir = os.path.dirname(os.path.abspath(__file__))
++xla_meta_path = file_dir + "/../xla/test/allowed_tests.py"
++xla_meta = run_path(xla_meta_path)
++assert xla_meta
++allowed_xla_tests = xla_meta.get('allowed_xla_tests', None)
++assert allowed_xla_tests is not None
++
++
 +class XLATestBase(DeviceTypeTestBase):
 +    device_type = 'xla'
-+
-+    allowed_tests = {
-+        'test_addcdiv',
-+        'test_addcmul',
-+        'test_diagonal',
-+        'test_isinf',
-+        'test_clamp',
```

Comment:
I fixed `temp_clamp` ;) This need to be added to the new list :D

Reply:
Thanks Ailing!